SleepEngine/Engine/Object.h

```diff
@@ -13,13 +13,12 @@ class Object
 
     Component* addComponent(ComponentsContainer::value_type&& component);
 
-    template <class TComponent>
-    TComponent* addComponent()
+    template <class TComponent, class ...Args>
+    TComponent* addComponent(Args&&... constructorArgs)
     {
-        static_assert(std::is_default_constructible_v <TComponent>, "TComponent is not default constructible. Please, "
-            "create an instance of this class and use non-templated 'addComponent' method.");
-        Component* handle = addComponent(std::make_unique <TComponent>());
-        return static_cast<TComponent*>(handle);
+		static_assert(std::is_base_of_v <Component, TComponent>, "TComponent should be a heir of slp::Component class!");
```

Comment:
descendant, not heir)

Reply:
![image](https://user-images.githubusercontent.com/23706483/56856027-e1eeaf80-695a-11e9-8883-64cc001505cd.png)

unless you've meant it as a Homestuck reference

Reply:
![image](https://user-images.githubusercontent.com/24576933/56856079-e10a4d80-695b-11e9-8dd6-e570183f5cec.png)
How about it, Elon Musk?